File: wiseSaying/src/main/java/wiseSayingController/WiseSayingController.java
package wiseSayingController;

import wiseSayingService.WiseSayingService;

import java.util.Scanner;

public class WiseSayingController {
    private final WiseSayingService service;
    private final Scanner sc;

    public WiseSayingController() {
        this.service = new WiseSayingService();
        this.sc = new Scanner(System.in);
    }

    public void start() {
        System.out.println("== Wise Saying App ==");

        while (true) {
            System.out.print("Command) ");
            String command = sc.nextLine().trim();

            switch (command) {
                case "register":
                    registerProverb();
                    break;
                case "list":
                    listProverb();
                    break;
                case "build":
                    buildDataProverb();
                    break;
                case "exit":
                    System.out.println("Program exited.");
                    return;
                default:
                    if (command.startsWith("search")) {
                        searchProverb(command);
                    } else if (command.startsWith("update?id=")) {
                        updateProverb(command);
                    } else if (command.startsWith("delete?id=")) {
                        deleteProverb(command);
                    } else {
                        System.out.println("Invalid command.");
                    }
                    break;
            }
        }
    }

    private void registerProverb() {
        // proverbList 에 등록한 명언 추가 -> service 단
        // 실제로 .json 파일 추가 -> repository 단

        System.out.print("명언 : ");
        String proverb = sc.nextLine();

        System.out.print("작가 : ");
        String author = sc.nextLine();

        service.registerProverb(proverb, author);
    }

    private void listProverb() {
        System.out.print("페이지 번호 입력: ");
        int page = Integer.parseInt(sc.nextLine());
        service.listProverbs(page);
    }

    private void searchProverb(String command) {
        String[] parts = command.split("\\?");

        if (parts.length < 2) {
            System.out.println("잘못된 검색입니다. ");
            return;
        }

        String[] keywords = parts[1].split("&");
        if (keywords.length < 2) {
            System.out.println("검색타입 또는 검색어가 누락되었습니다.");
            return;
        }

        String keywordType = keywords[0].split("=")[1];
        String keyword = keywords[1].split("=")[1];
        service.searchProverb(keywordType, keyword);
    }

    private void updateProverb(String command) {
        int id = Integer.parseInt(command.split("=")[1]);
        service.updateProverb(id, sc);
    }

    private void deleteProverb(String command) {
        int id = Integer.parseInt(command.split("=")[1]);
        service.deleteProverb(id);
    }

    private void buildDataProverb() {
        service.buildDataToFile();
    }
}

File: migrationWiseSaying/src/main/kotlin/controller/WiseSayingController.kt
package controller

import global.SingletonScope
import service.WiseSayingService

class WiseSayingController {

    private val service: WiseSayingService = SingletonScope.wiseSayingService

    fun start() {
        println("== Wise Saying App ==")

        while (true) {
            println("Command): ")
            val command: String = readlnOrNull()?.trim() ?: ""

            when (command) {
                "등록" -> registerProverb()
                "목록" -> listProverb()
                "빌드" -> buildDataProverb()
                "종료" -> println("Program exited!")
                else -> {
                    when {
                        command.startsWith("검색") -> searchProverb(command)
                        command.startsWith("수정?id=") -> updateProverb(command)
                        command.startsWith("삭제?id=") -> deleteProverb(command)
                        else -> println("Invalid command.")
                    }
                }
            }
        }
    }

    private fun registerProverb() {
        print("명언 : ")
        val proverb: String = readlnOrNull()?.trim() ?: ""

        print("작가 : ")
        val author: String = readlnOrNull()?.trim() ?: ""

        service.registerProverb(proverb, author)
    }

    private fun listProverb() {
        print("페이지 번호 입력: ")
        val page: Int = readlnOrNull()?.trim()?.toInt() ?: 1
        service.listProverbs(page)
    }

    private fun buildDataProverb() {
        service.buildDataToFile()
    }

    private fun searchProverb(command: String) {
//        val parts: MutableList<String> = command.split("\\?").toMutableList()
        val parts: List<String> = command.split("\\?")

        if (parts.size < 2) {
            println("잘못된 검색입니다.")
            return
        }

        val keywords: List<String> = parts[1].split("&")
        if (keywords.size < 2) {
            println("검색 타입 또는 검색어가 누락되었습니다.")
            return
        }

        val keywordType: String = keywords[0].split("=")[1]
        val keyword: String = keywords[1].split("=")[1]
        service.searchProverb(keywordType, keyword)
    }

    private fun updateProverb(command: String) {
        val id: Int = command.split("=")[1].toInt()
        service.updateProverb(id)
    }

    private fun deleteProverb(command: String) {
        val id: Int = command.split("=")[1].toInt()
        service.deleteProverb(id)
    }
}
